src/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs

```diff
@@ -792,21 +792,35 @@ public override string ToString()
         {
             if (_sddlForm == null)
             {
-                StringBuilder result = new StringBuilder();
-
                 //
                 // Typecasting of _IdentifierAuthority to a long below is important, since
                 // otherwise you would see this: "S-1-NTAuthority-32-544"
                 //
-
-                result.Append("S-1-").Append((long)_identifierAuthority);
-
-                for (int i = 0; i < SubAuthorityCount; i++)
+                // length of buffer calculation
+                // prefix = "S-1-".Length: 4;
+                // authority: long.MaxValue.ToString("D") : 19;
```

Comment:
It would be nice for this comment to be in sync with the code:
- This comment says `long`, code uses `ulong`
- This comment says `uint`, code uses `int`

I would be best to use unsigned types everywhere - both in the comment and in the code.

Reply:
Done, and changed the compatibility version to use ulong too.